### PR TITLE
Adding .NET SDK and tests for new RP - Managed Service Identity

### PR DESCRIPTION
**Pull request by varunkch**
## Description
This PR is for adding the autorest generated C# SDK and tests for the new RP - Managed Service Identity.
Swagger spec pull request: https://github.com/Azure/azure-rest-api-specs/pull/2020
---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [Done] Please add REST spec PR link to the SDK PR
- [Done] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [Done] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [Done] Title of the pull request is clear and informative.
- [Done] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [Done] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [Done] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/ManagedServiceIdentity/Management.ManagedServiceIdentity/Microsoft.Azure.Management.ManagedServiceIdentity.csproj [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  <!-- Please do not move/edit code below this line -->
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+  <!-- Please do not move/edit code below this line -->
+
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.Management.ManagedServiceIdentity</PackageId>
+    <Description>Provides developers with libraries for ManagedSericeIdentity under Azure Resource manager to perform CRUD operations on identities.</Description>
+    <Version>1.0.0.0-preview</Version>
+    <AssemblyName>Microsoft.Azure.Management.ManagedServiceIdentity</AssemblyName>
+    <PackageTags>Microsoft Azure resource management;msi;managedserviceidentity</PackageTags>
```
Comment (shahabhijeet) [inline src/SDKs/ManagedServiceIdentity/Management.ManagedServiceIdentity/Microsoft.Azure.Management.ManagedServiceIdentity.csproj]: add ; at end of the last tag

File: src/SDKs/ManagedServiceIdentity/ManagedServiceIdentity.Tests/ManagedServiceIdentity.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Management.ManagedServiceIdentity\Microsoft.Azure.Management.ManagedServiceIdentity.csproj" />
+  </ItemGroup>
+
+  <!--Do not remove until VS Test Tools fixes #472-->
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
+  </ItemGroup>
```
Comment (shahabhijeet) [inline src/SDKs/ManagedServiceIdentity/ManagedServiceIdentity.Tests/ManagedServiceIdentity.Tests.csproj]: @varunkch use wildcard to add all json files in your project. see an example
https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Compute/Compute.Tests/Compute.Tests.csproj#L20

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Can you do the following in the below order:
1)	Pull latest from upstream (if you have already done this good)
2)	Execute msbuild.exe build.proj from the repo root
a.	This will update the build tools in the current repo
3)	Generate your SDK
4)	Build you sdk at least one from command line of IDE
a.	Msbuild.exe build.proj /t:build /p:Scope=SDKs\CognitiveService\management
5)	Update the PR

**Comment by varunkch** [comment]
I am getting the following error on running msbuild.exe build.proj from the repo root:

"C:\Users\vakuma\Documents\GitHub\Forks\azure-sdk-for-net\build.proj" (default target) (1) ->
(GetLatestBuildTools target) ->
  C:\Users\vakuma\Documents\GitHub\Forks\azure-sdk-for-net\tools\bootstrapTools\bootstrap.targets(6,2): error : The remote server returned an error: (404) Not Found. [C:\Users\vakuma\Documents\GitHub\Forks\azure-sdk-for-net\build.proj]

    0 Warning(s)
    1 Error(s)

File: src/SDKs/ManagedServiceIdentity/ManagedServiceIdentity.Tests/ManagedServiceIdentity.Tests.csproj (hunk 2 above)
Comment (varunkch) [reply src/SDKs/ManagedServiceIdentity/ManagedServiceIdentity.Tests/ManagedServiceIdentity.Tests.csproj]: Done.

**Pull request closed without merging**